includes/Hooks.php

```diff
@@ -100,6 +103,11 @@ public function onHistoryTools(
 		$oldRevisionRecord,
 		$userIdentity
 	) {
+		// [UGC-4257] Don't show thank links if user doesn't have specific permission
+		if ( !ThanksPermissions::checkUserPermissionsForThanks( RequestContext::getMain()->getOutput() ) ) {
+			return;
+		}
+
```

Comment:
(non-blocking) This can be replaced with a hook invocation, that way Fandom code wouldn't be coupled with this class anymore. It's likely that the hook can be upstreamed, which will reduce the fork size in future versions.